libs/platform/src/lib/components/input-group/input-group.component.ts

```diff
@@ -0,0 +1,154 @@
+import {
+    Component,
+    Input,
+    ChangeDetectionStrategy,
+    ContentChildren,
+    QueryList,
+    ElementRef,
+    ViewEncapsulation,
+    ChangeDetectorRef,
+    Renderer2,
+    AfterViewInit,
+    OnInit,
+    AfterContentInit
+} from '@angular/core';
+import { startWith } from 'rxjs/operators';
+
+import { BaseComponent } from '../base';
+import { InputComponent } from '../form/input/input.component';
+import { ContentDensity, Status } from '../form/form-control';
+
+import { InputGroupConfig } from './input-group.config';
+import { InputGroupAddonComponent } from './addon/addon.component';
+
+const CSS_CLASS_NAME = {
+    inputGroup: 'fdp-input-group',
+    inputGroupInnerInput: 'fd-input-group__input'
+} as const;
+
+/**
+ * Fundamental input group component
+ *
+ * ```html
+ * <fdp-input-group>
+ *   <fdp-input-group-addon>$</fdp-input-group-addon>
+ *   <fdp-input type="number" name="price"></fdp-input>
+ * </fdp-input-group>
+ * ```
+ *
+ */
+@Component({
+    selector: 'fdp-input-group',
+    templateUrl: './input-group.component.html',
+    styleUrls: ['./input-group.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class InputGroupComponent extends BaseComponent implements OnInit, AfterContentInit, AfterViewInit {
+    /**
+     * content Density of element: 'cozy' | 'compact'
+     */
+    @Input()
+    set contentDensity(contentDensity: ContentDensity) {
+        this._contentDensity = contentDensity;
+
+        this._setInputControlsOptions();
+        this._setAddonControlsOptions();
+    }
+    get contentDensity(): ContentDensity {
+        return this._contentDensity;
+    }
+
+    /**
+     * control state: 'success' | 'error' | 'warning' | 'default' | 'information'
+     */
+    @Input()
+    set state(state: Status) {
+        this._controlStateClass = `is-${state}`;
+    }
+
+    /** @hidden */
+    @ContentChildren(InputComponent)
+    private _inputControls: QueryList<InputComponent>;
+
+    /** @hidden */
+    @ContentChildren(InputGroupAddonComponent)
+    private _addonControls: QueryList<InputGroupAddonComponent>;
+
+    /** @hidden */
+    private _contentDensity: ContentDensity = this._inputGroupConfig.contentDensity;
+
+    /** @hidden */
+    _controlStateClass: string;
+
+    /** @hidden */
+    constructor(
+        _cd: ChangeDetectorRef,
+        private _renderer: Renderer2,
+        protected _elementRef: ElementRef<HTMLElement>,
+        protected _inputGroupConfig: InputGroupConfig
+    ) {
+        super(_cd);
+    }
+
+    /** @hidden */
+    ngOnInit(): void {
+        /**
+         * Assign predefined className to host element
+         * to make css selector stronger
+         */
+        this._renderer.addClass(this._elementRef.nativeElement, CSS_CLASS_NAME.inputGroup);
+    }
+
+    /** @hidden */
+    ngAfterContentInit(): void {
+        this._inputControls.changes
```

Comment:
Never unsubscribed

Reply:
There is no need as angular does it for us https://github.com/angular/angular/blob/c01bd0fe8e0b65430d7b7893c937e50110ec0020/packages/core/src/linker/query_list.ts#L153-L156